packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -924,66 +941,65 @@ class _ViewContentState extends State<_ViewContent> {
-                              widget.viewOnChanged?.call(value);
-                              updateSuggestions();
-                            },
-                            onSubmitted: widget.viewOnSubmitted,
-                            textCapitalization: widget.textCapitalization,
-                            textInputAction: widget.textInputAction,
-                            keyboardType: widget.keyboardType,
-                          ),
-                        ),
-                      ),
-                      FadeTransition(
-                        opacity: viewDividerFadeCurve,
-                        child: viewDivider),
-                      Expanded(
-                        child: FadeTransition(
-                          opacity: viewListFadeOnIntervalCurve,
-                          child: viewBuilder(result),
-                        ),
+            child: FadeTransition(
+              opacity: viewIconsFadeCurve,
+              child: Column(
+                mainAxisSize: MainAxisSize.min,
+                crossAxisAlignment: CrossAxisAlignment.stretch,
+                children: <Widget>[
+                  Padding(
+                    padding: EdgeInsets.only(top: widget.topPadding),
+                    child: SafeArea(
+                      top: false,
+                      bottom: false,
+                      child: SearchBar(
+                        autoFocus: true,
+                        constraints: headerConstraints ?? (widget.showFullScreenView ? BoxConstraints(minHeight: _SearchViewDefaultsM3.fullScreenBarHeight) : null),
+                        leading: widget.viewLeading ?? defaultLeading,
+                        trailing: widget.viewTrailing ?? defaultTrailing,
+                        hintText: widget.viewHintText,
+                        backgroundColor: const MaterialStatePropertyAll<Color>(Colors.transparent),
+                        overlayColor: const MaterialStatePropertyAll<Color>(Colors.transparent),
+                        elevation: const MaterialStatePropertyAll<double>(0.0),
+                        textStyle: MaterialStatePropertyAll<TextStyle?>(effectiveTextStyle),
+                        hintStyle: MaterialStatePropertyAll<TextStyle?>(effectiveHintStyle),
+                        controller: _controller,
+                        onChanged: (String value) {
+                          widget.viewOnChanged?.call(value);
+                          updateSuggestions();
+                        },
+                        onSubmitted: widget.viewOnSubmitted,
+                        textCapitalization: widget.textCapitalization,
+                        textInputAction: widget.textInputAction,
+                        keyboardType: widget.keyboardType,
                       ),
-                    ],
+                    ),
                   ),
-                ),
+                  if (_controller.isOpen && (widget.showFullScreenView || result.isNotEmpty)) ...<Widget>[
```

Comment:
I don't think we need this condition check since this class itself is showing the view content right?

Reply:
I prefer this method of avoiding the assertion issue instead of adding another widget to the tree unless you have an objection

Reply:
I removed the check for _controller.isOpen. The rest of the check is to not show the divider when there are no results to show.